stl/inc/xstring

```diff
@@ -4269,55 +4298,62 @@ template <class _Elem, class _Traits, class _Alloc>
 _NODISCARD basic_string<_Elem, _Traits, _Alloc> operator+(
     const basic_string<_Elem, _Traits, _Alloc>& _Left, const basic_string<_Elem, _Traits, _Alloc>& _Right) {
     // return string + string
-    basic_string<_Elem, _Traits, _Alloc> _Ans;
-    _Ans.reserve(_Left.size() + _Right.size());
-    _Ans += _Left;
-    _Ans += _Right;
-    return _Ans;
+    if (_Left.max_size() - _Left.size() < _Right.size()) {
+        _Xlen_string();
+    }
+
+    return basic_string<_Elem, _Traits, _Alloc>(
+        _String_constructor_concat_tag{}, _Left, _Left.c_str(), _Left.size(), _Right.c_str(), _Right.size());
 }
 
 template <class _Elem, class _Traits, class _Alloc>
 _NODISCARD basic_string<_Elem, _Traits, _Alloc> operator+(
     _In_z_ const _Elem* const _Left, const basic_string<_Elem, _Traits, _Alloc>& _Right) {
-    using _String_type = basic_string<_Elem, _Traits, _Alloc>;
-    using _Size_type   = typename _String_type::size_type;
-    _String_type _Ans;
-    _Ans.reserve(_Convert_size<_Size_type>(_Traits::length(_Left) + _Right.size()));
-    _Ans += _Left;
-    _Ans += _Right;
-    return _Ans;
+    using _String_type    = basic_string<_Elem, _Traits, _Alloc>;
+    using _Size_type      = typename _String_type::size_type;
+    const auto _Left_size = _Convert_size<_Size_type>(_Traits::length(_Left));
+    if (_Right.max_size() - _Right.size() < _Left_size) {
+        _Xlen_string();
+    }
+
+    return basic_string<_Elem, _Traits, _Alloc>(

```

Comment:
You can use `_String_type` here.

Reply:
Actually going to eliminate string_type and apply the de-`explicit` you suggested in the other comment.